### PR TITLE
Visualization code two

### PR DESCRIPTION
**Pull request by kathleen-lund**
Data parsing functions, jest tests and a D3 code for a new bar chart visualization (currently with dummy data)

### CODE REVIEW

File: capstone/src/main/webapp/barVisualization.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>STEP Capstone</title>
+    <link rel="stylesheet" href="style.css" />
+  </head>
+  <body onload="createBarVisualization(); setLegend();">
+    <svg id="barChart" class="barChart" focusable="true"></svg>
```
Comment (Bickor) [inline capstone/src/main/webapp/barVisualization.html:9]: nit: Shouldn't the class names and ids be separated by a dash?

File: capstone/src/main/webapp/style.css [hunk 2]
```diff
@@ -13,6 +13,18 @@
   width: 800px;
 }
 
+.barChart {
```
Comment (Bickor) [inline capstone/src/main/webapp/style.css:16]: Same comment about the names.

Comment (kathleen-lund) [reply capstone/src/main/webapp/style.css:16]: Great catch, thanks! Fixing

File: capstone/src/main/webapp/visualizationdata.js [hunk 3]
```diff
@@ -208,11 +208,134 @@ function getNumWeeksBetween(date1, date2) {
...
+  // Iterate through all page view data for each restaurant
+  // and push average for each to return array
+  for (let i = 0; i < restaurantPageViews.length; i++) {
+    const currRestaurantPageViews = restaurantPageViews[i].pageViews;
+    let sum = 0;
+    for (let j = 0; j < currRestaurantPageViews.length; j++) {
+      sum += currRestaurantPageViews[j].count;
+    }
+    const count = currRestaurantPageViews.length;
+    const average = sum / count;
```
Comment (tzsebe) [inline capstone/src/main/webapp/visualizationdata.js:259]: is it possible for a restaurant to have no page views and cause this to div by zero?

File: capstone/src/main/webapp/barvisualization.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    transitionStacked();
+  } else {
+    transitionGrouped();
+  }
+}
+
+/**
+ * Simple function to set a legend describing the bar chart
+ */
+function setLegend() {
```
Comment (tzsebe) [inline capstone/src/main/webapp/barvisualization.js:146]: Per warning, should this be used somewhere?

**Pull request merged**